src/modules/LanItems.rb

```diff
@@ -1005,53 +1011,26 @@ def Read
 
       ReadHw()
       NetworkInterfaces.Read
+      NetworkInterfaces.adapt_old_config!
       NetworkInterfaces.CleanHotplugSymlink
 
       interfaces = getNetworkInterfaces
       # match configurations to Items list with hwinfo
-      Builtins.foreach(interfaces) do |confname|
-        pos = nil
-        val = {}
-        Builtins.foreach(
-          Convert.convert(
-            @Items,
-            from: "map <integer, any>",
-            to:   "map <integer, map <string, any>>"
-          )
-        ) do |key, value|
-          if Ops.get_string(value, ["hwinfo", "dev_name"], "") == confname
-            pos = key
-            val = deep_copy(value)
-          end
-        end
-        if pos.nil?
-          pos = Builtins.size(@Items)
-          Ops.set(@Items, pos, {})
+      interfaces.each do |confname|
+        @Items.each do |key, value|
+          match = value.fetch("hwinfo", {}).fetch("dev_name", "") == confname
+          @Items[key]["ifcfg"] = confname if match
         end
-        Ops.set(@Items, [pos, "ifcfg"], confname)
```

Comment:
Previously we always set the "ifcfg" attribute for Items with hwinfo and also for Items without it, so the loop below does nothing at the end.

So we could keep the method above just with some cleanup, or this alternative that only sets the ifcfg attribute as the interface name if match, and makes the loop below useful.